File: app/src/main/java/com/lifedawn/bestweather/findaddress/map/adapter/FavoriteLocationItemViewPagerAdapter.java
package com.lifedawn.bestweather.findaddress.map.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lifedawn.bestweather.R;
import com.lifedawn.bestweather.databinding.FoundAddressItemViewpagerBinding;
import com.lifedawn.bestweather.findaddress.map.interfaces.OnClickedLocationBtnListener;
import com.lifedawn.bestweather.findaddress.map.interfaces.OnClickedScrollBtnListener;
import com.lifedawn.bestweather.room.dto.FavoriteAddressDto;

import java.util.List;

public class FavoriteLocationItemViewPagerAdapter extends RecyclerView.Adapter<FavoriteLocationItemViewPagerAdapter.FavoriteItemViewHolder> {
	private List<FavoriteAddressDto> addressList;
	private OnClickedLocationBtnListener<FavoriteAddressDto> onClickedLocationBtnListener;
	private OnClickedScrollBtnListener onClickedScrollBtnListener;
	private boolean showAddBtn = true;

	public FavoriteLocationItemViewPagerAdapter setShowAddBtn(boolean showAddBtn) {
		this.showAddBtn = showAddBtn;
		return this;
	}

	public FavoriteLocationItemViewPagerAdapter setAddressList(List<FavoriteAddressDto> addressList) {
		this.addressList = addressList;
		return this;
	}

	public FavoriteLocationItemViewPagerAdapter setOnClickedLocationBtnListener(OnClickedLocationBtnListener<FavoriteAddressDto> onClickedLocationBtnListener) {
		this.onClickedLocationBtnListener = onClickedLocationBtnListener;
		return this;
	}

	public FavoriteLocationItemViewPagerAdapter setOnClickedScrollBtnListener(OnClickedScrollBtnListener onClickedScrollBtnListener) {
		this.onClickedScrollBtnListener = onClickedScrollBtnListener;
		return this;
	}

	public List<FavoriteAddressDto> getAddressList() {
		return addressList;
	}

	@NonNull
	@Override
	public FavoriteItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		return new FavoriteItemViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.found_address_item_viewpager, parent, false));
	}

	@Override
	public void onBindViewHolder(@NonNull FavoriteItemViewHolder holder, int position) {
		holder.setDataView();
	}

	@Override
	public int getItemCount() {
		return addressList.size();
	}


	class FavoriteItemViewHolder extends RecyclerView.ViewHolder {
		protected FoundAddressItemViewpagerBinding binding;

		public FavoriteItemViewHolder(@NonNull View view) {
			super(view);
			binding = FoundAddressItemViewpagerBinding.bind(view);

			binding.addBtn.setVisibility(showAddBtn ? View.VISIBLE : View.GONE);

			binding.addBtn.setOnClickListener(new View.OnClickListener() {
				@Override
				public void onClick(View view) {
					onClickedLocationBtnListener.onSelected(addressList.get(getBindingAdapterPosition()), false);
				}
			});

			binding.removeBtn.setOnClickListener(new View.OnClickListener() {
				@Override
				public void onClick(View view) {
					onClickedLocationBtnListener.onSelected(addressList.get(getBindingAdapterPosition()), true);
				}
			});

			binding.left.setOnClickListener(new View.OnClickListener() {
				@Override
				public void onClick(View v) {
					onClickedScrollBtnListener.toLeft();
				}
			});

			binding.right.setOnClickListener(new View.OnClickListener() {
				@Override
				public void onClick(View v) {
					onClickedScrollBtnListener.toRight();
				}
			});
		}

		void setDataView() {
			final int position = getBindingAdapterPosition();
			String itemPosition = (position + 1) + " / " + getItemCount();
			binding.itemPosition.setText(itemPosition);

			FavoriteAddressDto address = addressList.get(position);

			binding.addressName.setText(address.getAddress());
			binding.country.setText(address.getCountryName());

		}
	}
}

File: app/src/main/java/com/lifedawn/bestweather/ui/findaddress/map/adapter/FavoriteLocationItemViewPagerAdapter.kt
package com.lifedawn.bestweather.ui.findaddress.map.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.lifedawn.bestweather.R
import com.lifedawn.bestweather.data.local.room.dto.FavoriteAddressDto
import com.lifedawn.bestweather.databinding.FoundAddressItemViewpagerBinding
import com.lifedawn.bestweather.ui.findaddress.map.adapter.FavoriteLocationItemViewPagerAdapter.FavoriteItemViewHolder
import com.lifedawn.bestweather.ui.findaddress.map.interfaces.OnClickedLocationBtnListener
import com.lifedawn.bestweather.ui.findaddress.map.interfaces.OnClickedScrollBtnListener

class FavoriteLocationItemViewPagerAdapter : RecyclerView.Adapter<FavoriteItemViewHolder>() {
    var addressList: List<FavoriteAddressDto>? = null
        private set
    private var onClickedLocationBtnListener: OnClickedLocationBtnListener<FavoriteAddressDto>? = null
    private var onClickedScrollBtnListener: OnClickedScrollBtnListener? = null
    private var showAddBtn = true
    fun setShowAddBtn(showAddBtn: Boolean): FavoriteLocationItemViewPagerAdapter {
        this.showAddBtn = showAddBtn
        return this
    }

    fun setAddressList(addressList: List<FavoriteAddressDto>?): FavoriteLocationItemViewPagerAdapter {
        this.addressList = addressList
        return this
    }

    fun setOnClickedLocationBtnListener(onClickedLocationBtnListener: OnClickedLocationBtnListener<FavoriteAddressDto>?): FavoriteLocationItemViewPagerAdapter {
        this.onClickedLocationBtnListener = onClickedLocationBtnListener
        return this
    }

    fun setOnClickedScrollBtnListener(onClickedScrollBtnListener: OnClickedScrollBtnListener?): FavoriteLocationItemViewPagerAdapter {
        this.onClickedScrollBtnListener = onClickedScrollBtnListener
        return this
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavoriteItemViewHolder {
        return FavoriteItemViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.found_address_item_viewpager, parent, false))
    }

    override fun onBindViewHolder(holder: FavoriteItemViewHolder, position: Int) {
        holder.setDataView()
    }

    override fun getItemCount(): Int {
        return addressList!!.size
    }

    inner class FavoriteItemViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        protected var binding: FoundAddressItemViewpagerBinding

        init {
            binding = FoundAddressItemViewpagerBinding.bind(view)
            binding.addBtn.visibility = if (showAddBtn) View.VISIBLE else View.GONE
            binding.addBtn.setOnClickListener { onClickedLocationBtnListener!!.onSelected(addressList!![bindingAdapterPosition], false) }
            binding.removeBtn.setOnClickListener { onClickedLocationBtnListener!!.onSelected(addressList!![bindingAdapterPosition], true) }
            binding.left.setOnClickListener { onClickedScrollBtnListener!!.toLeft() }
            binding.right.setOnClickListener { onClickedScrollBtnListener!!.toRight() }
        }

        fun setDataView() {
            val position = bindingAdapterPosition
            val itemPosition = (position + 1).toString() + " / " + itemCount
            binding.itemPosition.text = itemPosition
            val address = addressList!![position]
            binding.addressName.text = address.displayName
            binding.country.text = address.countryName
        }
    }
}